lib/ReactViews/DragDropFile.tsx

```diff
@@ -97,72 +85,65 @@ class DragDropFile extends Component<PropsType> {
+      runInAction(() => (viewState.isDraggingDroppingFile = false));
     } catch (e) {
-      props.viewState.terria.raiseErrorToUser(e, "Failed to upload files");
+      viewState.terria.raiseErrorToUser(e, "Failed to upload files");
     }
-  }
+  };
 
-  @action
-  handleDragEnter(e: DragEvent) {
+  const handleDragEnter = action((e: DragEvent) => {
     e.preventDefault();
     e.stopPropagation();
     e.dataTransfer.dropEffect = "copy";
-    this.target = e.target;
-  }
+    targetRef.current = e.target;
+  });
 
-  handleDragOver(e: DragEvent) {
+  const handleDragOver = (e: DragEvent) => {
     e.preventDefault();
-  }
+  };
 
-  @action
-  handleDragLeave(e: MouseEvent) {
+  const handleDragLeave = action((e: MouseEvent) => {
     e.preventDefault();
     if (e.screenX === 0 && e.screenY === 0) {
-      this.props.viewState.isDraggingDroppingFile = false;
+      viewState.isDraggingDroppingFile = false;
     }
-    if (e.target === document || e.target === this.target) {
-      this.props.viewState.isDraggingDroppingFile = false;
+    if (e.target === document || e.target === targetRef.current) {
+      viewState.isDraggingDroppingFile = false;
     }
-  }
-
-  @action
-  handleMouseLeave() {
-    this.props.viewState.isDraggingDroppingFile = false;
-  }
-
-  render() {
-    return (
-      <div
-        onDrop={this.handleDrop.bind(this)}
-        onDragEnter={this.handleDragEnter.bind(this)}
-        onDragOver={this.handleDragOver.bind(this)}
-        onDragLeave={this.handleDragLeave.bind(this)}
-        onMouseLeave={this.handleMouseLeave.bind(this)}
-        className={classNames(Styles.dropZone, {
-          [Styles.isActive]: this.props.viewState.isDraggingDroppingFile
-        })}
-      >
-        {this.props.viewState.isDraggingDroppingFile ? (
-          <div className={Styles.inner}>
-            <Trans i18nKey="dragDrop.text">
-              <h3 className={Styles.heading}>Drag & Drop</h3>
-              <div className={Styles.caption}>
-                Your data anywhere to view on the map
-              </div>
-            </Trans>
-          </div>
-        ) : (
-          ""
-        )}
-      </div>
-    );
-  }
-}
-
-export default withTranslation()(withViewState(DragDropFile));
+  });
+
+  const handleMouseLeave = action(() => {
+    viewState.isDraggingDroppingFile = false;
+  });
+
+  return (
+    <div
+      onDrop={handleDrop}
+      onDragEnter={handleDragEnter}
+      onDragOver={handleDragOver}
+      onDragLeave={handleDragLeave}
+      onMouseLeave={handleMouseLeave}
+      className={classNames(Styles.dropZone, {
+        [Styles.isActive]: viewState.isDraggingDroppingFile
+      })}
+    >
+      {viewState.isDraggingDroppingFile ? (
+        <div className={Styles.inner}>
+          <Trans i18nKey="dragDrop.text">
+            <h3 className={Styles.heading}>Drag & Drop</h3>
+            <div className={Styles.caption}>
+              Your data anywhere to view on the map
```

Comment:
Since you seem to be the expert on translation, should this text have a translation?

Reply:
It is translated there is a `Trans` component around it